doc/rst/tools/chplcheck/chplcheck.rst

```diff
@@ -299,6 +300,73 @@ For example, the following defines a rule that has a fixit associated with it:
            return BasicRuleResult(node, fixits=[fixit])
        return True
 
+.. note::
+
+    The fixit for 'NoFunctionFoo' demonstrated here is not production-ready.
+    It do not rename the uses of the function, nor does it check for conflicts
+    with other names in the file. It is intended only to demonstrate the API for
+    defining fixits. The same is true for various other versions of this
+    example in this section.
+
+
+The above code snippet directly uses a ``fixits=`` argument to the
+``BasicRuleResult`` constructor. This is one of two ways to associate fixits with
+rules. Both advanced and basic rule results support this constructor argument.
+
+Constructing the ``Edit`` objects can become relatively complicated for more
+powerful auto-fixes. In practice, ``chplcheck`` developers have observed
+that the code to construct the necessary edits can be longer than the code
+implementing the rule. To provide a separation of concerns between
+"what should be warned about" and "how to fix the warning", ``chplcheck``
+provides a second mechanism to attach fixits to rules: the ``@driver.fixit``
+decorator.
+
+This decorator accepts the rule-to-be-fixed as an argument, and wraps a function
+that accepts the result of the rule. The wrapped function should use
+the information returned by the rule to construct a ``Fixit`` object or a list of
+``Fixit`` objects.
+
+For example, the snippet above can be written using the decorator as follows:
+
+.. code-block:: python
+
+   @driver.basic_rule(chapel.Function)
+   def NoFunctionFoo(context, node):
+       return node.name() != "foo"
+
+   @driver.fixit(NoFunctionFoo)
+   def FixNoFunctionFoo(context, result):
```

Comment:
I think this example would be clearer if you are explicit about the type of `result`. It is implied by the surrounding paragraphs and the variable name, but I think it could be more explicit. Consider adding a type annotation or mentioning it in the above paragraph